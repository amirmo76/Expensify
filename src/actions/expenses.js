import uuid from "uuid";

export const addExpense = ({
  description = "",
  note = "",
  amount = 0,
  createdAt = 0
} = {}) => ({
  type: "ADD_EXPENSE",
  expense: {
    id: uuid(),
    description,
    note,
    amount,
    createdAt
  }
});

export const removeExpense = ({ id } = {}) => {
  if (!id) {
    throw new Error("not specifying the expense id to remove!");
  }

  return {
    type: "REMOVE_EXPENSE",
    id
  };
};

export const editExpense = ({ id, data } = {}) => {
  if (!id) {
    throw new Error("not specifying the expense id to edit!");
  }

  return {
    type: "EDIT_EXPENSE",
    id,
    data
  };
};
