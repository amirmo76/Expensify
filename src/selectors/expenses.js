export const getVisibleExpenses = (
  expenses,
  { text, sortBy, startDate, endDate }
) => {
  return expenses
    .filter(cur => {
      const textMatch = cur.description
        .toLowerCase()
        .includes(text.toLowerCase());

      const startDateMatch =
        typeof startDate !== "number" || cur.createdAt >= startDate;

      const endDateMatch =
        typeof endDate !== "number" || cur.createdAt <= endDate;

      return textMatch && startDateMatch && endDateMatch;
    })
    .sort((a, b) => {
      if (sortBy === "amount") {
        return b.amount - a.amount;
      }
      return b.createdAt - a.createdAt;
    });
};
