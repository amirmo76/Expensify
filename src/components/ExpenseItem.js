import React from 'react'
import { connect } from 'react-redux';
import { removeExpense } from "../actions/expenses";

const ExpenseItem = ({ description, amount, createdAt, id, dispatch }) => (
  <div>
    <p>Description: {description}, amount: {amount}, created at: {createdAt}</p>
    <button onClick={e => {
      dispatch(removeExpense({ id }))
    }}>Remove</button>
  </div>
)

export default connect()(ExpenseItem);