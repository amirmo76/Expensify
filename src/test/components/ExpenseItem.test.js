import React from "react";
import expenses from "../fixtures/expenses";
import ExpenseItem from "../../components/ExpenseItem";
import { shallow } from "enzyme";

test("should render ExpenseItem", () => {
  const wrapper = shallow(<ExpenseItem {...expenses[0]} />);
  expect(wrapper).toMatchSnapshot();
});
