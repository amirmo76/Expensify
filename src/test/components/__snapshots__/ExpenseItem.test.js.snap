// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseItem 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h3>
      Coffee
    </h3>
  </Link>
  <p>
     
    500
     - 
    -172800000
     
  </p>
</div>
`;
